agent/structs/structs.go

```diff
@@ -519,6 +519,7 @@ type ServiceSpecificRequest struct {
 	Datacenter      string
 	NodeMetaFilters map[string]string
 	ServiceName     string
+	ServiceKind     ServiceKind
```

Comment:
Does adding this here in this generic struct impact any other users of this struct?

Reply:
I don't think so, callers/APIs that don't need it will just ignore. Like the Connect/Ingress booleans that were added below.